packages/sirius-web/backend/sirius-web/src/test/java/org/eclipse/sirius/web/services/migration/MigrationParticipantTests.java

```diff
@@ -0,0 +1,200 @@
+                }
+                if (nodeDescription.getStyle() instanceof ImageNodeStyleDescription) {
+                    assertThat(nodeDescription.getOutsideLabels().get(0).getLabelExpression()).isEqualTo("aql:'NodeWithImage'");
+                }
+                //Test that UpdateNodeDescriptionDomainTypeFeatureName updated the feature domainType name correctly
+                assertThat(nodeDescription.getDomainType()).isEqualTo("flow::CompositeProcessor");
+            });
+
+            //Test that when serialized, the document is updated with the last MigrationParticipant metadata
+            Optional<DocumentData> document = resourceToDocumentService.toDocument(jsonResource);
+            assertThat(document).isPresent();
+            document.ifPresent(documentData -> {
+                var optionalMigrationData = resourceLoader.getMigrationDataFromDocumentContent(documentData.document().getContent());
+                assertThat(optionalMigrationData).isPresent();
+                optionalMigrationData.ifPresent(migrationData -> {
+                    assertThat(migrationData.getMigrationVersion()).isEqualTo(UpdateDiagramDescriptionNameValue.PARTICIPANT_VERSION);
+                    assertThat(migrationData.getLastMigrationPerformed()).isEqualTo(UpdateDiagramDescriptionNameValue.class.getSimpleName());
+                });
+
+            });
+        }
+    }
+
+    private static class UpdateLabelExpressionMigrationParticipant implements IMigrationParticipant {
+
+        private static final String PARTICIPANT_VERSION = "2024.4.0";
+
+        @Override
+        public String getVersion() {
+            return PARTICIPANT_VERSION;
+        }
+
+        @Override
+        public void getJSonObjectBeforeLoading(EObject eobject, JsonObject object) {
+            if (eobject instanceof NodeDescription nodeDescription) {
+               var nodeDescriptionData = object.getAsJsonObject("data");
+               var labelExpression = nodeDescriptionData.get("labelExpression").getAsString();
+               if (nodeDescription.getStyle() instanceof RectangularNodeStyleDescription) {
+                   InsideLabelDescription insideLabelDescription = DiagramPackage.eINSTANCE.getDiagramFactory().createInsideLabelDescription();
+                   insideLabelDescription.setLabelExpression(labelExpression);
+                   nodeDescription.setInsideLabel(insideLabelDescription);
+               }
+                if (nodeDescription.getStyle() instanceof ImageNodeStyleDescription) {
+                    OutsideLabelDescription outsideLabelDescription = DiagramPackage.eINSTANCE.getDiagramFactory().createOutsideLabelDescription();
+                    outsideLabelDescription.setLabelExpression(labelExpression);
+                    nodeDescription.getOutsideLabels().add(outsideLabelDescription);
+                }
+            }
+        }
```

Comment:
Is that the migration participent for the NodeDescription#labelExpression which is now missing? That should be directly in the studio services of `sirius-web-application`